src/main/java/picard/sam/BamErrorMetric/CollectBamErrorMetrics.java

```diff
@@ -0,0 +1,464 @@
+        if (superValidation != null) errors.addAll(Arrays.asList(superValidation));
+        if (!errors.isEmpty()) {
+            return errors.toArray(new String[errors.size()]);
+        } else {
+            return null;
+        }
+    }
+
+    private final Log log = Log.getInstance(CollectBamErrorMetrics.class);
+
+    @Override
+    protected int doWork() {
+
+        final Random random = new Random(42);
+
+        final ProgressLogger progressLogger = new ProgressLogger(log, 100000);
+        long nTotalLoci = 0;
+        long nSkippedLoci = 0;
+        long nProcessedLoci = 0;
+
+        IOUtil.assertFileIsReadable(INPUT);
+        IOUtil.assertFilesAreReadable(INTERVALS);
+
+        final Collection<BaseErrorAggregation> aggregatorList = getAggregatorList();
+        // Open up the input resources:
+        try (
+                final SamReader sam = SamReaderFactory.makeDefault().open(INPUT);
+                final ReferenceSequenceFileWalker referenceSequenceFileWalker = new ReferenceSequenceFileWalker(REFERENCE_SEQUENCE);
+                final PeekableIterator<VariantContext> vcfIterator = new PeekableIterator<>(
+                    VCF == null ? Collections.emptyIterator() : new VCFFileReader(VCF, true).iterator())
+        ) {
+
+            final SAMSequenceDictionary sequenceDictionary = referenceSequenceFileWalker.getSequenceDictionary();
+            if (sam.getFileHeader().getSortOrder() != SAMFileHeader.SortOrder.coordinate)
+                throw new PicardException("Input BAM must be sorted by coordinate");
+
+            sequenceDictionary.assertSameDictionary(sam.getFileHeader().getSequenceDictionary());
+
+            final IntervalList regionOfInterest = getIntervals(sequenceDictionary);
+
+            log.info("Getting SamLocusIterator");
+
+            final SamLocusIterator samLocusIterator = new SamLocusIterator(sam, regionOfInterest);
+
+            // Now go through and compare information at each of these sites
+            samLocusIterator.setEmitUncoveredLoci(false);
+            samLocusIterator.setMappingQualityScoreCutoff(MIN_MAPPING_Q);
+            samLocusIterator.setQualityScoreCutoff(MIN_BASE_Q);
+
+            log.info("Using " + aggregatorList.size() + " aggregators.");
+
+            aggregatorList.forEach(la ->
+                    IOUtil.assertFileIsWritable(new File(OUTPUT + la.getSuffix())));
+
+            // iterate over loci
+            log.info("Starting iteration over loci");
+
+            final SAMLocusAndReferenceIterator iterator = new SAMLocusAndReferenceIterator(referenceSequenceFileWalker, samLocusIterator);
+
+            // This hasNext() call has side-effects. It loads up the index and makes sure that the iterator is really ready for
+            // action. Calling this allows for the logging to be more accurate.
+            iterator.hasNext();
+            log.info("Really starting iteration now.");
```

Comment:
How about changing this and the previous log message to be "preparing traversal" and "starting traversal" or something similar?